intro-to-R-tidyverse/02-intro_to_tidyverse.Rmd

```diff
@@ -252,34 +252,86 @@ We can use the same function, but instead use the `desc` function too.
 metadata %>% arrange(desc(age_mths))
 ``` 
 
-What if we want to obtain summary statistics based on groups? 
-Conveniently named, there's a function called `group_by` that seamlessly allows
-us to do this. 
+What if we would like to create new variables? 
+For that we use `mutate` function. 
+Let's convert the ages that are in months to years and make that into a new variable.
+Here we are also using the `round` function from base R to limit the number of 
+decimal points in our new variable to 2.
+
+```{r}
+metadata %>% 
+  mutate(age_yrs = round(age_mths/12, 2))
+```
+
+What if we want to obtain summary statistics for a variable? 
+The `summarize` function allows us to calculate summary statistics for a column. 
+Here we will use summarize to obtain an overall mean age in months. 
+
+```{r}
+metadata %>% summarize(mean(age_mths))
+```
+
+What if we'd like to obtain a summary statistics but have them for various groups?
+Conveniently named, there's a function called `group_by` that seamlessly 
+allows us to do this. 
 Also note that `group_by` allows us to group by multiple variables at a time if 
 you want to.
 
 ```{r}
-metadata %>% group_by(sex) %>% 
-             summarize(mean(age_mths))
+metadata_summary <- metadata %>%
+      group_by(sex) %>% 
+      summarize(mean(age_mths))
 ```
 
-What if we would like to create new variables based on other variables? 
-For that we use `mutate` function. 
-Let's convert the ages that are in months to years and make that into a new variable.
-Here we are also using the `round` function from base R to limit the number of 
-decimal points in our new variable to 2.
+Let's use `head(metadata_summary)` so we can look at a preview of what we made:
+
+```{r}
+head(metadata_summary)
+```
+
+## A brief intro to the `apply` family of functions
+
+In the previous section, we used the `summarize` section to calculate summary
+statistics for variables. 
+However, in base R there the `apply` family of functions can be an alternative 
+methods for performing calculations or transformations across a data.frame,
+matrix or for other object structures. 
+
+We will practice using `apply` here. 
 
 ```{r}
-metadata_edit <- metadata %>% 
-                 mutate(age_yrs = round(age_mths/12, 2))
+# Make a matrix for apply practice
```

Comment:
You introduce `set.seed` in the notebook right before this. Including it here could be an opportunity to reiterate that lesson.

Reply:
Good idea!